go/libraries/doltcore/sqle/dfunctions/dolt_commit.go

```diff
@@ -0,0 +1,224 @@
+// Copyright 2020 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dfunctions
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/dolthub/go-mysql-server/sql"
+
+	"github.com/dolthub/dolt/go/cmd/dolt/cli"
+	"github.com/dolthub/dolt/go/libraries/doltcore/env/actions"
+	"github.com/dolthub/dolt/go/libraries/doltcore/sqle"
+	"github.com/dolthub/dolt/go/libraries/utils/argparser"
+)
+
+const DoltCommitFuncName = "dolt_commit"
+
+type DoltCommitFunc struct {
+	children []sql.Expression
+}
+
+// NewDoltCommitFunc creates a new DoltCommitFunc expression.
+func NewDoltCommitFunc(args ...sql.Expression) (sql.Expression, error) {
+	return &DoltCommitFunc{children: args}, nil
+}
+
+const (
+	allowEmptyFlag   = "allow-empty"
+	dateParam        = "date"
+	commitMessageArg = "message"
+	forceFlag        = "force"
+	authorParam      = "author"
+)
+
+func createArgParser() *argparser.ArgParser {
```

Comment:
Same comment as Andy made about duplicating logic. Share the arg parser used by the cmd package.